packages/notifi-react/lib/context/NotifiTargetContext.tsx

```diff
@@ -257,23 +317,62 @@ export const NotifiTargetContextProvider: FC<PropsWithChildren> = ({
     [],
   );
 
-  const renewTargetGroup = useCallback(() => {
-    setIsLoading(true);
-    return frontendClient
-      .ensureTargetGroup(targetGroupToBeSaved)
-      .then((_result) => {
-        frontendClient
-          .fetchData()
-          .then((data) => {
-            refreshTargetDocument(data);
-            setError(null);
-          })
-          .catch((e) => setError(e as Error))
-          .finally(() => setIsLoading(false));
-      })
-      .catch((e) => setError(e as Error))
-      .finally(() => setIsLoading(false));
-  }, [frontendClient, targetGroupToBeSaved]);
+  const renewTargetGroup = useCallback(
+    async (singleTargetRenewArgs?: TargetRenewArgs) => {
+      let data = { ...targetGroupToBeSaved };
+
+      if (singleTargetRenewArgs) {
+        const { target, value } = singleTargetRenewArgs;
+        if (isFormTargetRenewArgs(singleTargetRenewArgs)) {
+          let formTarget: string | undefined = '';
+          let formValue: string | undefined = '';
+
+          if (target === 'email') {
+            formTarget = 'emailAddress';
+            formValue = value === '' ? undefined : value;
+          }
+
+          if (target === 'telegram') {
+            formTarget = 'telegramId';
+            formValue =
+              value === '' ? undefined : formatTelegramForSubscription(value);
+          }
+
+          if (target === 'phoneNumber') {
+            formTarget = target;
+            formValue = isValidPhoneNumber(value) ? value : undefined;
+          }
+
+          data = {
+            ...targetGroupToBeSaved,
+            [formTarget]: formValue,
+          };
```

Comment:
Just being curious about this section.
As far as I know, this block only happens when we want to update `singleTarget`. But I think others will get updated if we use destruct `targetGroupToBeSaved` in which there might have some changed target?
Or do I miss anything?

Reply:
@ankush-notifi FYI

Reply:
You're correct, Eric. However, if we don't  pass in `targetGroupToBeSaved` directly, the frontendClient.ensureTargetGroup function will remove existing target groups.

Reply:
Ah ... what about using the data from `targetData` instead of `targetGroupToBeSaved` as `targetData` reflects the data stored in BE. wdyt?